### DISCUSSION TITLE
Sample loop ping-pong

### DISCUSSION THREAD

**Discussion by o010b** [SynthstromAudible/DelugeFirmware#1384] [Ideas]
Playing samples looping in ping-pong mode is an old school classic to have clickless looping for sustaining sounds without having to fiddle with single sample loop points back and forth to try find a sweet spot. 

That is - sample traversal going back and forth from loop points rather than just wrap over from loop end to loop start. 

```
Simple
               Start         Loop Start       Loop End          End           
                  |              |               |                |
Note On           |----------------------------->|                |
Looping                          |-------------->|                |
Note Off                         |------------------------------->|

---

Ping-Pong
                Start         Loop Start       Loop End          End           
                  |              |               |                |
Note On           |----------------------------->|                |
                                 |<--------------|                |
Looping                          |-------------->|                |
                                 |<--------------|                |
Note Off                         |------------------------------->|

```
It would open up for much easier live work with samples and resampling on the device rather than doing it in a sample editor and transfer back and forth to the Deluge.

**Comment by lobit12** [comment]
I would add auto-zero crossing detection to this too.  Would help alot

**Reply by o010b** [reply]
PIng-pong wouldn't be affected by zero crossing. It just reverses playback at loop end point so there is no clicking.

**Comment by ok-reza** [comment]
solid idea!

**Comment by worker-gray** [comment]
similar to 'zig-zag' playback mentioned here:
https://github.com/SynthstromAudible/DelugeFirmware/discussions/663

**Reply by worker-gray** [reply]
I would add that additional control/automation for SPEED of sample playback would be a great enhancement.
